src/dmd/func.d

```diff
@@ -3618,3 +3620,21 @@ extern (C++) final class DeleteDeclaration : FuncDeclaration
         v.visit(this);
     }
 }
+
+void showArgMismatch(Loc loc, Expressions* fargs, TypeFunction tf, size_t failIndex)
+{
+    if (failIndex < fargs.dim && failIndex < tf.parameters.dim)
+    {
+        auto arg = (*fargs)[failIndex];
+        auto par = (*tf.parameters)[failIndex];
+        const rv = (!arg.isLvalue() && par.storageClass & (STCref | STCout)) ? "rvalue " : "";
+        const pc = parameterToChars(par, tf.varargs);
+        const ts = toAutoQualChars(arg.type, par.type);
+        auto msg = "cannot pass %sargument `%s` of type `%s` to parameter `%s`";
+        // don't print parameter type if it's already in the parameter string
+        if (strcmp(par.type.toChars(), ts[1]) != 0)
+            msg ~= " of type `%s`";
```

Comment:
This isn't guaranteed to produce a 0 terminated array.

Reply:
Just use two different `msg` strings.

Reply:
i.e.:
```
msg = strcmp() ? "cannot pass ..." : "cannot pass ... of type ...";
```

Reply:
Fixed by adding `\0` to 2nd string, hope that's OK?